src/ruby/ext/grpc/rb_call.c

```diff
@@ -235,7 +235,7 @@ static VALUE grpc_rb_call_set_metadata(VALUE self, VALUE metadata) {
 */
 static int grpc_rb_md_ary_fill_hash_cb(VALUE key, VALUE val, VALUE md_ary_obj) {
   grpc_metadata_array *md_ary = NULL;
-  int array_length;
+  long array_length;
   int i;
```

Comment:
This counter is incremented up to `array_length` and used as the index argument to `rb_ary_entry`. Would you mind changing this one as well?